torch/_meta_registrations.py

```diff
@@ -5398,6 +5398,77 @@ def meta_polygamma(n: int, self: Tensor) -> Tensor:
     return torch.empty_like(self, dtype=result_dtype)
 
 
+def _create_unary_float_meta_func(func):
+    @register_meta(func)
+    @out_wrapper()
+    def _f(x):
+        return _elementwise_meta(
+            x, type_promotion=ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.INT_TO_FLOAT
+        )
+
+    return _f
+
+
+def _create_binary_float_meta_func(func):
+    @register_meta(func)
+    @out_wrapper()
+    def _f(x, y):
+        x, y = _maybe_broadcast(x, y)
+        return _elementwise_meta(
+            x, y, type_promotion=ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.INT_TO_FLOAT
+        )
+
+    return _f
```

Comment:
_elementwise_meta is mainly for prims, but also already used in this file. I've considered inlining relevant function calls instead (broadcasting, etc.), but it's less code with this helper.

Reply:
In core:
- `CREATE_UNARY_FLOAT_META_FUNC` (wrapper around `build_borrowing_unary_float_op`)
- and `build_borrowing_binary_float_op`.